### PR TITLE
run e2e-openstack-extended test in self hosted runner

### PR DESCRIPTION
**Pull request by eslutsky**
this GH action will trigger  openstack extended tests in  a self hosted runner .

this PR consists of the following items:
-  [x] trigger the test  with a certain comment and report the job status on the PR.
-  [ ] provision self hosted fedora VM on ovirt and join as a runner to forklift gh repo.
    - [x] create a fedora template  with kind cluster +  gh-runner instance pre-configured.
            each template is configured to work with specific `org/repo`
    - [x] create a VM from the template (limited to 1  VM per template)
    - [ ] allow running several VM instances from same template - generate runner-id and reconfigure the runner 
-  [x] run the extended on the self hosted runner.
-  [x] remove the runner VM
-  [x] make sure only [codeowner](https://github.com/kubev2v/forklift/blob/main/.github/CODEOWNERS#L5) in `tests/ can trigger the job using  comment.
-  [ ] scale-out - support multiple  concurrent runs (up to maximum)

### CODE REVIEW

**Comment by eslutsky** [comment]
example [PR](https://github.com/eslutsky/forklift/pull/4) that show this functionality

File: .github/workflows/create-new-pr.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - id: codeowner
+        uses: SvanBoxel/codeowners-action@v1
+        with:
+          path: './.github/CODEOWNERS'
+
+      - name: fail if the commenter doesnt exists in codeowners
+        id: allowedaccess
+        run: |
+          echo allowed=$(echo '${{ steps.codeowner.outputs.codeowners }}' | \
+          jq  -r '."tests/" | index("@${{ github.actor }}")') >> $GITHUB_OUTPUT
```
Comment (ahadas) [inline .github/workflows/create-new-pr.yml:24]: isn't it too strict? I'm ok with letting developers that are not code-owners to initiate the test, how about checking it in a similar way to [how it's done for oVirt's tests](https://github.com/oVirt/ovirt-system-tests/blob/master/.github/workflows/ost.yaml#L37-L38)?

File: .github/workflows/run-osp-extended-tests.yml [hunk 2]
```diff
@@ -0,0 +1,151 @@
+name: openstack-extended-tests
+on:
+  workflow_dispatch:
+    name:
+      description: 'run e2e openstack  extended tests on Self hosted runner'
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:5]:
```suggestion
      description: 'run e2e openstack extended tests on a self-hosted runner'
```

File: .github/workflows/run-osp-extended-tests.yml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+      - id: codeowner
+        uses: SvanBoxel/codeowners-action@v1
+        with:
+          path: './.github/CODEOWNERS'
+
+      - name: fail if the commenter doesnt exists in codeowners
+        run: |
+          [[ $(echo '${{ steps.codeowner.outputs.codeowners }}' | \
+          jq  -r '."tests/" | index("@${{ github.event.comment.user.login }}")') -eq 1 ]]
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:43]: the other comment on this check should have been placed here but also applies there to the message by the bot

File: .github/workflows/run-osp-extended-tests.yml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        if: ${{ failure() || cancelled() }}
+        with:
+          sha: ${{ steps.comment-branch.outputs.head_sha }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+
+  ci_setup:
+    if: github.event.issue.pull_request && contains(github.event.comment.body, '/test-osp-extended')
+    runs-on: self-hosted
+    #runs-on: ubuntu-latest
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:65]: remove then?

File: .github/workflows/run-osp-extended-tests.yml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+          path: forklift
+
+      - name: Checkout forkliftci
+        uses: actions/checkout@v3
+        with:
+          repository: kubev2v/forkliftci
+          ref: v5.0
+
+      - name: add docker-hub token
+        run: echo "${{ secrets.DOCKER_HUB_TOKEN }}" | docker login --username "${{  secrets.DOCKER_HUB_USER }}" --password-stdin
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:84]: why do we need docker hub?

File: .github/workflows/run-osp-extended-tests.yml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+          sha: ${{ steps.comment-branch.outputs.head_sha }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+
+      - name: Add comment to PR
+        uses: actions/github-script@v6
+        if: always()
+        with:
+          script: |
+            const name = '${{ github.workflow   }}';
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:126]:
```suggestion
            const name = '${{ github.workflow }}';
```

File: .github/workflows/run-osp-extended-tests.yml (hunk 5 above)
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml:84]: it needed for pulling kind images , with out this line it fails on docker Pull rate limits.

File: .github/workflows/create-new-pr.yml (hunk 1 above)
Comment (eslutsky) [reply .github/workflows/create-new-pr.yml]: refactored

File: .github/workflows/run-osp-extended-tests.yml (hunk 5 above)
Comment (ahadas) [reply .github/workflows/run-osp-extended-tests.yml:80]: ack

File: .github/workflows/create-new-pr.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    if: github.event.issue.pull_request &&
+      startsWith(github.event.comment.body, '/test-osp-extended') && (
+      github.event.comment.author_association == 'MEMBER' ||
+      github.event.comment.author_association == 'COLLABORATOR')
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout forklift
+        uses: actions/checkout@v3
```
Comment (ahadas) [inline .github/workflows/create-new-pr.yml:17]: do we really need to checkout the forklift repo here?

File: .github/workflows/run-osp-extended-tests.yml [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+          sha: ${{ steps.comment-branch.outputs.head_sha }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: pending
+
+      - name: Checkout forklift
+        uses: actions/checkout@v3
+
+      - name: prepare the secrets
+#        uses: kubev2v/forkliftci/ci/prepare-ansible-secrets@v5.0
+        uses: eslutsky/forkliftci/ci/prepare-ansible-secrets@validate-ssh-key
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:43]: will be changed before merging?

File: .github/workflows/run-osp-extended-tests.yml [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        with:
+          OKD_ENGINE_SECRETS: ${{ secrets.OKD_ENGINE_SECRETS }}
+          OKD_SSH_KEY: ${{ secrets.OKD_SSH_KEY }}
+
+      - name: provision self hosted runner
+        #uses: kubev2v/forkliftci/ci/create-self-runner@v5.0
+        uses: eslutsky/forkliftci/ci/create-self-runner@validate-ssh-key
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:50]: same here

File: .github/workflows/run-osp-extended-tests.yml [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+          suite_name: e2e-sanity-openstack-extended
+
+      - name: save k8s logs and upload-artifact
+        if: ${{ always() }}
+        uses: kubev2v/forkliftci/ci/save-artifacts@v5.0
+        with:
+          source_provider: openstack
+
+      - name: Get PR branch
+        if: always() && github.event_name == 'issue_comment'
```
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:105]: why doing this rather than just `if: github.event_name == 'issue_comment'`? (here and below)

**Review by ahadas — COMMENTED** [review]
looks good overall, few minor comments inside

File: .github/workflows/create-new-pr.yml (hunk 7 above)
Comment (eslutsky) [reply .github/workflows/create-new-pr.yml]: apparently not, fixed.

File: .github/workflows/run-osp-extended-tests.yml (hunk 8 above)
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml]: done

File: .github/workflows/run-osp-extended-tests.yml (hunk 9 above)
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml]: done

File: .github/workflows/run-osp-extended-tests.yml (hunk 10 above)
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml:103]: we want this action to be triggered regardless of the previous step result (succees or failure),
according to the [docs](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions) 
```
You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions.
```
Where “one of these functions” refers to:
- sucess()
- always()
- cancelled()
- failure()

Comment (ahadas) [reply .github/workflows/run-osp-extended-tests.yml:103]: got it

**Review by ahadas — APPROVED** [review]

**Pull request merged**